rpc/src/responses/block_ids_in_chain.rs

```diff
@@ -19,6 +20,66 @@ impl Response for BlocksInChainResponse {
 #[cfg(test)]
 mod test {
     use super::*;
+    use serde::{Deserialize, Serialize};
+
+    /// Mock struct that represents the
+    /// object expected to be found in
+    /// a response.
+    #[derive(Serialize, Deserialize)]
+    struct MockBlockInChain {
+        block_id: String,
+    }
+
+    struct BlocksInChainMockResponse {
+        pub mock_block_ids: json_array::JsonArray<MockBlockInChain>,
+    }
+
+    impl Response for BlocksInChainMockResponse {
+        /// Parses a response string in the form
+        /// `"[[{"block_id":"alpha_numeric_block_id_string"}], ["..."]]"` or
+        /// `"[{"block_id":"alpha_numeric_block_id_string"}, ...]"` into a
+        /// [`BlocksInChainMockResponse`](Self).
+        fn from_response_str(response: &str) -> Result<Self, ParseError> {
+            let mock_block_ids = json_array::JsonArray::from_str(response)?;
+            Ok(Self { mock_block_ids })
+        }
+    }
+
+    #[test]
+    fn get_blocks_in_chain_from_object_response_expecting_string_fails() {
+        let mock_response = r#"[[{"block_id": "1"}]]"#;
+
+        let blocks_response = BlocksInChainResponse::from_response_str(mock_response);
+        assert!(blocks_response.is_err());
+    }
```

Comment:
Not really sure _what_ this is testing, is it just testing out that you can't parse this format of `mock_response` into a `struct BlocksInChainResponse`? If so, this seems to do more harm than good from a cognitive load standpoint.

I had to re-read the method a couple of times to see why this worked but the next test passed, which might indicate that the difference between `BlockInChainResponse` and `BlocksInChainMockResponse` is too small, they are very easy to look over and read as the same.

Either this test module should only test `BlocksInChainMockResponse` or we should find a better way to write tests with a mock struct.

Reply:
This test would probably make more sense if it was being tested in the JSON array file or in a response file where the RPC response is of the form `[[{"block_id": "1"}]]`, consider it removed!